File: readability.py
from cs50 import get_string

text = get_string("Text: ")

word_count = 0
letter_count = 0
sentence_count = 0
a = ".!?"

for n in text:
    if n == " ":
        word_count += 1
    if n.isalpha() == True:
        letter_count += 1
    if n in a:
        sentence_count += 1

word_count += 1

L = (letter_count / word_count) * 100
S = (sentence_count / word_count) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index > 15:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    int space_count = 0;
    int letter_count = 0;
    int sentence_count = 0;
    string a = ".!?";

    string input = get_string("Text: ");

    int length = strlen(input);
    for (int i = 0; i < length; i++)
    {
        if (isalnum(input[i]) != 0)
        {
            letter_count++;
        }

        if (isspace(input[i]) != 0)
        {
            space_count++;
        }

        if (input[i] == a[0] || input[i] == a[1] || input[i] == a[2])
        {
            sentence_count++;
        }
    }

    int words = space_count + 1;

    float letter_average = letter_count / (float) words * 100;

    float sentence_average = sentence_count / (float) words * 100;

    float index = 0.0588 * letter_average - 0.296 * sentence_average - 15.8;

    int index_round = round(index);

    // print grade of the text under condition
    if (index_round < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index_round > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index_round);
    }
}
